redaxo/src/core/lib/login/user.php

```diff
@@ -176,7 +176,8 @@ public function hasPerm($perm)
      * @param string $key Complex perm key
      *
      * @return rex_complex_perm|null Complex perm
-     * @psalm-return rex_media_perm|rex_structure_perm|rex_module_perm|rex_clang_perm|null
+     * @psalm-return rex_complex_perm|null
+     * @phpstan-return rex_media_perm|rex_structure_perm|rex_module_perm|rex_clang_perm|null
```

Comment:
Auch phpstan mecker über diesen type.. siehe https://github.com/redaxo/redaxo/pull/3374/files

ggf hier psalm und phpstan return entfernen?

Reply:
Du hast ihn ja extra für phgstan eingeführt: https://github.com/redaxo/redaxo/commit/3c3fa9e7863bfa3f87736961bf555ac73f4186a9

Ansonsten gäbe es an vielen Stellen Fehler, weil die Methoden nicht gefunden werden.

In Psalm gibt es die Fehler nicht, weil psalm sich die richtige Klasse je nach $key-Parameter aus der .phpstorm.meta.php holt.

Mir gefällt an der Typangabe aber auch nicht, dass sie eigentlich nicht stimmt. Es können auch noch andere Klassen rauskommen, yform und ytraduko liefern zum Beispiel eigene complex-perms.

Reply:
Ok müssen wir mal schauen. Im aktuellen level gibts hier aktuell probleme